internal/pkg/database/migrations.go

```diff
@@ -0,0 +1,32 @@
+package database
+
+import (
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+)
+
+var migrations = []migration{
+	{
+		name: "1_users_create_table",
+		fn: func(db Queryer) error {
+			_, err := db.Exec("CREATE TABLE IF NOT EXISTS users (email text, password text)")
+			return err
+		},
+	}, {
+		name: "2_users_add_email_uniqueness",
+		fn: func(db Queryer) error {
+			_, err := db.Exec("ALTER TABLE users ADD UNIQUE (email);")
+			return err
+		},
+	},
+}
+
+func ApplyMigrations(db *sqlx.DB) error {
+	err := migrate(db, migrations)
```

Comment:
As you're including a number at the start, maybe sort the array before applying?

Reply:
Hmm, I think the main purpose is that they are added incrementally, that way you can see the order of the applied migrations. It could have been a timestamp, but that is more difficult to obtain rather than just adding 1 to the previous migration. Conflicts when merging are more clear with numbers I think.

The main point is, new migrations should be added to the bottom, and never should migrations be inserted in the middle.